### PR TITLE
searchable container implementation

### PR DESCRIPTION
**Pull request by bellybubjoeOfficial**
# Description

We created a reusable search bar component and replaced uses of "TileGrid" in several files with the newly defined component. The choices made in re-factoring (eg. the addElementComponent prop) should be reviewed and suggestions are welcome.

## Issues

No issue

## Screenshots

UI should be the same

## Test

In the files where we replaced the components, open the corresponding page and see that it is p much identical (hopefully).

### CODE REVIEW

File: src/pages/admin-home.tsx [hunk 1]
```diff
@@ -12,7 +12,6 @@ import React, {
   SetStateAction,
```
Comment (nickbar01234) [inline src/pages/admin-home.tsx:12]: We can revert changes to this file as it's part of the old implementation.

**Review by nickbar01234 — CHANGES_REQUESTED** [review]
This is a great first start. I think we can add some changes to your implementation. Feel free to let me know if you disagree with any points!

1. Thanks for removing the old implementation; I think we can revert the changes here, since we won't be working with `pages/` directory, since it belongs to the old implementation.
2. In `SearchableContainer`, can we utilize `CardGrid` component to display the items? Is there a reason we need to support flexible CSS overriding here?
3. Also in `SearachableContainer`, shouldn't we display `SearchBar` too, since it belongs to the same abstraction. My thought was that `search()` should accept the text that is input into the search bar.

File: src/pages/admin-home.tsx (hunk 1 above)
Comment (cledi01) [reply src/pages/admin-home.tsx:12]: The changes made to this file is just for re-factoring to the new SearchableContainer component. Let me know if you were referring to something else 🙏

**Comment by cledi01** [comment]
2. The searchable container was largely replacing the tilegrid component, which was a singular div with the classname I put inside of the searchable container class.
3. I moved the searchbar inside and made a change to the format of the search prop function, pls double check 🙏 . Some of the card views don't support any searching (viewing the seniors a student is responsible for, viewing a seniors' files), so I also conditional rendering for the search bar to only appear, if a valid search function is passed in as props.

**Comment by cledi01** [comment]
Finalize changes on searchablecontainer to match what Nick desires... We didn't undo the changes made on the old implementation of admin-home to have an example on the codebase on how the SearchableContainer component can be used. Let us know if this is not needed and we can just undo the changes on that as well.

**Review by nickbar01234 — APPROVED** [review]
Great work! I added a small tweak to `display()` to accept index of the element, in case the client somehow needed it (maybe for `key`?).

**Pull request merged**